modules/slimfastq/main.nf

```diff
@@ -0,0 +1,52 @@
+def VERSION = '2.04'
+
+process SLIMFASTQ {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::slimfastq=2.04" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/slimfastq:2.04--h87f3376_2':
+        'quay.io/biocontainers/slimfastq:2.04--h87f3376_2' }"
+
+    input:
+    tuple val(meta), path(fastq)
+
+    output:
+    tuple val(meta), path("*.sfq"), emit: sfq
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if (meta.single_end) {
+        """
+        gzip -d -c '${fastq}' | slimfastq \\
+            $args \\
+            -f '${prefix}.sfq'
```

Comment:
Wouldn't it be "prettier" to chop of the `fastq.gz` extension and use the `SimpleName` from the original file?
I especially see an issue with the paired data  and the `_1` suffix. For example, at CMGG all data is standardized with an `_R1/_R2` suffix according to Illumina standard. This would break the naming convention for us (and others I imagine)

Reply:
`simpleName` is not great, in my opinion, because if your filename contains dots, it will happily chop away all of them. More importantly, I thought it was nf-core convention to use the prefix in output names so, yes, I can strip away extensions but I should rely on prefix as far as I know.

Reply:
By `SimpleName` I meant stripping of the the `{.fq,.fastq,.fq.gz,.fastq.gz}` extensions, possibly by regex or whatever.
I'm aware of the convention, but I still think it's dangerous and might cause issues down the line for users.

Anyway, I'm not very opinionated atm, since I'm not planning to use it, but I like future proofing.